packages/@react-aria/overlays/src/useOverlay.ts

```diff
@@ -43,7 +43,13 @@ export interface AriaOverlayProps {
    * out interaction with elements that should not dismiss the overlay.
    * By default, onClose will always be called on interaction outside the overlay ref.
    */
-  shouldCloseOnInteractOutside?: (element: Element) => boolean
+  shouldCloseOnInteractOutside?: (element: Element) => boolean,
+
+  /**
+   * Whether the overlay should be closed even if it isn't the top most visible overlay in the stack.
+   * Specifically used for submenu actions.
+   */
+  isMenu?: boolean
```

Comment:
Prop name to discuss if this ends up being the final solution we land on, still digging for alternative methods of closing all submenus when the user interacts outside

Reply:
well, i don't like making it specific to menus in case there is some other use case, but I'm also coming up with bad names, haha
isNestedDismissable, isChainDismissable
maybe make use of the non modal?

Reply:
same, having a hard time with the names too heh. `isRootDismissable`? 

I didn't really want to tie `isNonModal` to this behavior since `isNonModal` comes with a bunch of other behavior that might not be applicable to all cases where we want to close the root overlay regardless of the other nested overlays. I do see your point that the `isMenu` name might be too specific though...

Not sure what other overlays this would ever apply to though (Dialogs/Modals/default Popovers should stick to their current behavior of only closing the outermost overlay). Maybe `UNSTABLE_isMenu` 😅 ?

Reply:
no, i don't think it should be unstable, and that's still tied to menu. root dismissable is closer to what we mean, but it can also be, not the root. maybe, `isAlwaysDismissable`, or `isGroupDismissable`?

Reply:
I was thinking that we'd only apply this value to the root overlay and then it would close all sub overlays but I guess we could potentially have a subset of overlays have this behavior but not the root one, not sure what use case there is for that though. 

`isAlwaysDismissable` sounds good to me, we can add a good description to go along with it to clarify its usage similar to `isNonModal`. I'll make the change if we get group consensus.